src/pages/workspace/tags/WorkspaceTagsPage.tsx

```diff
@@ -0,0 +1,144 @@
+    const policyTagLists = useMemo(() => PolicyUtils.getTagLists(policyTags), [policyTags]);
+    const tagList = useMemo<PolicyForList[]>(
+        () =>
+            policyTagLists
+                .map((policyTagList) =>
+                    Object.values(policyTagList.tags).map((value) => ({
+                        value: value.name,
+                        text: value.name,
+                        keyForList: value.name,
+                        isSelected: !!selectedTags[value.name],
+                        rightElement: (
+                            <View style={styles.flexRow}>
+                                <Text style={[styles.disabledText, styles.alignSelfCenter]}>
+                                    {value.enabled ? translate('workspace.common.enabled') : translate('workspace.common.disabled')}
+                                </Text>
+                                <View style={[styles.p1, styles.pl2]}>
+                                    <Icon
+                                        src={Expensicons.ArrowRight}
+                                        fill={theme.icon}
+                                    />
+                                </View>
+                            </View>
+                        ),
+                    })),
+                )
+                .flat(),
+        [policyTagLists, selectedTags, styles.alignSelfCenter, styles.disabledText, styles.flexRow, styles.p1, styles.pl2, theme.icon, translate],
+    );
+
+    const toggleTag = (tag: PolicyForList) => {
+        setSelectedTags((prev) => ({
+            ...prev,
+            [tag.value]: !prev[tag.value],
+        }));
+    };
+
+    const toggleAllTags = () => {
+        const isAllSelected = tagList.every((tag) => !!selectedTags[tag.value]);
+        setSelectedTags(isAllSelected ? {} : Object.fromEntries(tagList.map((item) => [item.value, true])));
+    };
+
+    const getCustomListHeader = () => (
+        <View style={[styles.flex1, styles.flexRow, styles.justifyContentBetween, styles.pl3, styles.pr9]}>
+            <Text style={styles.searchInputStyle}>{translate('common.name')}</Text>
+            <Text style={[styles.searchInputStyle, styles.textAlignCenter]}>{translate('statusPage.status')}</Text>
+        </View>
+    );
+
+    return (
+        <AdminPolicyAccessOrNotFoundWrapper policyID={route.params.policyID}>
+            <PaidPolicyAccessOrNotFoundWrapper policyID={route.params.policyID}>
+                <ScreenWrapper
+                    includeSafeAreaPaddingBottom={false}
+                    style={[styles.defaultModalContainer]}
+                    testID={WorkspaceTagsPage.displayName}
+                    shouldShowOfflineIndicatorInWideScreen
+                >
+                    <HeaderWithBackButton
+                        icon={Illustrations.Tag}
+                        title={translate('workspace.common.tags')}
+                        shouldShowBackButton={isSmallScreenWidth}
+                    />
```

Comment:
Hey, I have a question related to this change. Every workspace page uses the `WorkspacePageWithSections` wrapper except for this one. Is this intentional?

Reply:
I don't think the new pages do, e.g. WorkspaceMembersPage, WorkspaceCategoriesPage

Reply:
The `WorkspaceMembersPage` copies the logic from the `WorkspacePageWithSections`.

I'm using this wrapper to fix multiple not found views, [and part of the PR is making the above page use the wrapper](https://github.com/Expensify/App/pull/36409/files#diff-e1b326f5a2dc9985458c8372d0b5dac4ed8ce5609cf65b3a30756b74dea8badf).

So what is the aim of the `WorkspacePageWithSections` wrapper? When should it be used?

Reply:
Hmm interesting case, maybe we should be using `WorkspacePageWithSections` here too and on all other pages. I think we can update these, but maybe in a separate PR.

Reply:
We probably have to think it over as currently, we have three different wrappers for workspaces, and we don't have clear guidelines on when to use which one.

Reply:
Yea, that makes sense. I'll create an issue for it.

Reply:
Created issue - https://github.com/Expensify/App/issues/37898

Reply:
Those wrappers also break the not-found views.

Going to e.g. https://staging.new.expensify.com/workspace/dssaddssdaads/categories` we see duplicated not found (that's what is the scope of the other issue) and after pressing the back button we see another ones...